Classes/common/HTTP/CDTHTTPInterceptorContext.m

```diff
@@ -0,0 +1,40 @@
+//
+//  CDTHTTPInterceptorContext.m
+//  
+//
+//  Created by Rhys Short on 17/08/2015.
+//
+//
+//  Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+//  except in compliance with the License. You may obtain a copy of the License at
+//    http://www.apache.org/licenses/LICENSE-2.0
+//  Unless required by applicable law or agreed to in writing, software distributed under the
+//  License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+//  either express or implied. See the License for the specific language governing permissions
+//  and limitations under the License.
+//
+
+#import "CDTHTTPInterceptorContext.h"
+
+@implementation CDTHTTPInterceptorContext
+
+// turn off the nonnull warning, we are passing null to cause a failure,
+// trying to use init should cause a compiler error anyway
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wnonnull"
+-(instancetype)init {
+    return [self initWithRequest:nil];
```

Comment:
I think calling into the designated initialiser method is the right thing to do, to ensure correct chaining. The designated initialiser already handles the case where the request is nil. @mikerhodes thoughts?